netlogo-gui/src/main/app/codetab/SmartIndenter.scala

```diff
@@ -33,167 +84,151 @@ extends Indenter {
   def handleCloseBracket() = handleTab()
 
   /// private helpers
-
-  private val TAB_WIDTH = 2
-  private def indentLine(lineNum: Int) {
-    val currentLine = code.getLineOfText(lineNum)
-    for(newSpaces <- computeNewSpaces(currentLine, lineNum)) {
-      val oldSpaces = countLeadingSpaces(currentLine)
-      if(newSpaces != oldSpaces) {
-        val lineStart = code.lineToStartOffset(lineNum)
-        if(newSpaces > oldSpaces)
-          code.insertString(lineStart, spaces(newSpaces - oldSpaces))
-        else
-          code.remove(lineStart, oldSpaces - newSpaces)
+  private def executeIndentations(indentations: Seq[LineIndent]): Unit = {
+    val start = indentations.head.lineStart
+    val finish = indentations.last.lineEnd
+    val builder = new StringBuilder(finish - start)
+    val offset = indentations.foldLeft(0) { (offset: Int, indent: LineIndent) =>
+      val newOffset = (offset, indent) match {
+        case (offset, AddIndent(lineNum, lineStart, _, text, delta)) =>
```

Comment:
This and the subsequent matches can be matched with something like `AddIndent(_, _, _, text, delta)`, can't they?